casanovo/utils.py

```diff
@@ -66,3 +75,221 @@ def split_version(version: str) -> Tuple[str, str, str]:
+) -> None:
+    """
+    Log general run report
+
+    Parameters
+    ----------
+    start_time : Optional[int], default=None
+        The start time of the sequencing run in seconds since the epoch.
+    end_time : Optional[int], default=None
+        The end time of the sequencing run in seconds since the epoch.
+    """
+    logger.info("======= End of Run Report =======")
+    if (start_time is not None) and (end_time is not None):
+        end_time = time.time() if end_time is None else end_time
+        start_datetime = datetime.fromtimestamp(start_time)
+        end_datetime = datetime.fromtimestamp(end_time)
+        delta_datetime = end_datetime - start_datetime
+        logger.info(
+            "Run Start Timestamp: %s",
+            start_datetime.strftime("%y/%m/%d %H:%M:%S"),
+        )
+        logger.info(
+            "Run End Timestamp: %s", end_datetime.strftime("%y/%m/%d %H:%M:%S")
+        )
+        logger.info("Time Elapsed: %s", delta_datetime)
+
+    logger.info("Executed Command: %s", " ".join(sys.argv))
+    logger.info("Executed on Host Machine: %s", socket.gethostname())
+
+    if torch.cuda.is_available():
+        gpu_util = torch.cuda.max_memory_allocated()
+        logger.info("Max GPU Memory Utilization: %dMiB", gpu_util >> 20)
+
+
+def log_sequencing_report(
+    predictions: Tuple[str, Tuple[str, str], float, float, float, float, str],
+    start_time: Optional[int] = None,
+    end_time: Optional[int] = None,
+    score_bins: List[float] = SCORE_BINS,
+) -> None:
+    """
+    Log sequencing run report
+
+    next_prediction : Tuple[
+        str, Tuple[str, str], float, float, float, float, str
+    ]
+        PSM predictions
+    start_time : Optional[int], default=None
+        The start time of the sequencing run in seconds since the epoch.
+    end_time : Optional[int], default=None
+        The end time of the sequencing run in seconds since the epoch.
+    score_bins: List[float], Optional
+        Confidence scores for creating confidence score distribution,
+        see get_score_bins
+    """
+    log_run_report(start_time=start_time, end_time=end_time)
+    run_report = get_report_dict(
+        pd.DataFrame(
+            {
+                "sequence": [psm[0] for psm in predictions],
+                "score": [psm[2] for psm in predictions],
+            }
+        ),
+        score_bins=score_bins,
+    )
+
+    if run_report is None:
+        logger.warning(
+            "No predictions were logged, this may be due to an error"
+        )
+    else:
+        num_spectra = run_report["num_spectra"]
+        logger.info("Sequenced %s spectra", num_spectra)
+        logger.info("Score Distribution:")
+        for score, pop in sorted(run_report["score_bins"].items()):
+            logger.info(
+                "%s spectra (%.2f%%) scored >= %s",
+                pop,
+                pop / num_spectra * 100,
+                score,
```

Comment:
**nitpick:** You can format the score as a float so that it has a consistent number of digits.